src/test/shell/bazel/bazel_coverage_java_test.sh

```diff
@@ -24,36 +24,34 @@ source "${CURRENT_DIR}/coverage_helpers.sh" \
   || { echo "coverage_helpers.sh not found!" >&2; exit 1; }
 
 
+RULES_JAVA_REPO_NAME=$(cat "$(rlocation io_bazel/src/test/shell/bazel/RULES_JAVA_REPO_NAME)")
```

Comment:
The mechanism used to override the remote Java and coverage tools didn't work with `--enable_bzlmod`, so I rewrote it borrowing some existing tricks like this one.

@meteorcloudy Could you review the changes to the test files (except for the new test case, which is unrelated to Bzlmod)?

Reply:
LGTM